### PR TITLE
Enable E2E tests on kubernetes/test-infra

### LINKED ISSUE #62: integrate e2e test with kubernetes/test-infra
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

could refer to:
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/cloud-provider-azure/cloud-provider-azure-config.yaml
https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes-sigs

WIP PR: https://github.com/kubernetes/test-infra/pull/12386
**Describe alternatives you've considered**

**Additional context**

/assign @ashishranjan738

### PR DESCRIPTION
**Pull request by chewong**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature
> /kind flake

/kind cleanup

**What this PR does / why we need it**:
This PR enables the project to grab the correct credentials to run E2E tests locally / on Prow. Before the tests begin, it will build the project, package it into a Docker image, and push it to a registry (defined through the env car `REGISTRY`). After that, the docker image will be installed into the test cluster using Helm.

**Which issue(s) this PR fixes**:

Fixes #62

**Special notes for your reviewer**:

**Release note**:
```

```

### CODE REVIEW

**Comment by chewong** [comment]
Travis CI is failing because I've removed `sudo` from some commands (`sudo` is not available when testing in Prow). I will open a separate PR to remove Travis CI configuration file after we have completely moved to Prow.
I've tested this PR with the upstream Prow instance and all tests (including E2E) passed. Here are the logs:

| Test type   | Logs                                                                                                          |
|-------------|---------------------------------------------------------------------------------------------------------------|
| Verify      | http://prow.upstream.azure-containers.io/log?job=pull-azurefile-csi-driver-verify&id=1176251696070791171      |
| Unit        | http://prow.upstream.azure-containers.io/log?job=pull-azurefile-csi-driver-unit&id=1176251696070791168        |
| Sanity      | http://prow.upstream.azure-containers.io/log?job=pull-azurefile-csi-driver-sanity&id=1176251696070791169      |
| Integration | http://prow.upstream.azure-containers.io/log?job=pull-azurefile-csi-driver-integration&id=1176251696070791170 |
| E2E         | http://prow.upstream.azure-containers.io/log?job=pull-azurefile-csi-driver-e2e&id=1176280636797751296         |

**Comment by andyzhangx** [comment]
@chewong can you also use this PR to switch to Prow. If you use another PR to fix this issue, then there could be a period of time when CI is always failing. Thanks.

File: test/e2e/README.md [hunk 1]
```diff
@@ -2,11 +2,21 @@
...
+export resourceGroup=<the resource group that your cluster is under>
+export location=<the location of your resource group>
 ```
-make e2e-test
+
+To run the E2E tests:
+
+```bash
+docker login # Login to docker so the test image can be pushed to your Docker Hub
+REGISTRY=<your Docker Hub ID> make e2e-test
```
Comment (andyzhangx) [inline test/e2e/README.md]: how can I use `mcr.microsoft.com/k8s/csi/azurefile-csi:version` to run e2e test?

Comment (chewong) [reply test/e2e/README.md]: I've made some adjustments to Makefile. You can now do that by executing `REGISTRY=mcr.microsoft.com/k8s/csi IMAGE_VERSION=version make e2e-test`.

**Comment by chewong** [comment]
Opened a PR on test-infra https://github.com/kubernetes/test-infra/pull/14650 to fix CI failure.

**Comment by chewong** [comment]
pull-azurefile-csi-driver-e2e should work after kubekins-e2e image is updated.

**Comment by chewong** [comment]
@andyzhangx all tests are green. Could you review this PR? Thanks.

**Pull request merged**